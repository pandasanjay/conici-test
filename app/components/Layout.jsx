import React from "react";
import Header from "./Header";
import Footer from "./Footer";

const Layout = ({ children }) => {
  return (
    <div>
      <Header title="Smart Check-out" />
      <div className="container">{children}</div>
      <Footer copywrite="@ 2018 - Hotel Beacons GmbH" />
    </div>
  );
};

export default Layout;
