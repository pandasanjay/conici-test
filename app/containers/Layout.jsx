import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import Header from "../components/Header";
import Footer from "../components/Footer";
import { connect } from "react-redux";
import { globalContentAction } from "../actions";
export class Layout extends Component {
  constructor(props) {
    super(props);
  }
  componentDidMount() {
    this.unlisten = this.props.history.listen(this.handelRouterEnter);
    this.handelRouterEnter();
  }
  componentWillUnmount() {
    this.unlisten();
  }
  handelRouterEnter(location, action) {
    this.props.globalContentAction();
  }
  render() {
    const title =
      this.props.globalReducer.pageContents &&
      this.props.globalReducer.pageContents.global.title;
    const copyright =
      this.props.globalReducer.pageContents &&
      this.props.globalReducer.pageContents.global.copyright;
    return (
      <div>
        <Header title={title} />
        <div className="container">{this.props.children}</div>
        <Footer copyright={copyright} />
      </div>
    );
  }
}
/*
 * mapDispatchToProps
 */
const mapDispatchToProps = dispatch => ({
  globalContentAction: () => dispatch(globalContentAction())
});

/*
 * mapStateToProps
 */
const mapStateToProps = state => ({
  ...state
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(Layout));
