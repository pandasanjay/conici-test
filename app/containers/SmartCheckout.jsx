import React, { Component } from "react";
import { connect } from "react-redux";
import { billingAddressAction, selectAddressAction } from "../actions";
import "./SmartCheckout.scss";
import ComponentLoader from "../components/ComponentLoader";
import BillingAddress from "../components/BillingAddressSection";

export class SmartCheckout extends Component {
  constructor(props) {
    super(props);
  }
  componentDidMount() {
    this.props.getBillingAddress();
  }
  render() {
    const { heading, description, components } =
      (this.props.globalReducer.pageContents &&
        this.props.globalReducer.pageContents.content) ||
      {};

    const { back, submit } =
      (this.props.globalReducer.pageContents &&
        this.props.globalReducer.pageContents.global) ||
      {};

    const addressList = this.props.billingAddressReducer.address || [];

    return (
      <div>
        <h2 className="text-center pt-3">{heading}</h2>
        <p className="text-center">{description}</p>
        {addressList.map((data, i) => (
          <BillingAddress
            {...data}
            key={i}
            id={i}
            onSelect={this.props.selectAddress}
          />
        ))}
        <ComponentLoader name="add_new_address" data={components || {}} />
        <div className="sh-actions d-flex mt-5 justify-content-between pt-2 pb-2">
          <div className="pt-2 pb-2">
            <button
              type="button"
              className="btn btn-outline-primary pl-5 pr-5 text-uppercase"
            >
              {back}
            </button>
          </div>
          <div className="pt-2 pb-2">
            <button
              type="button"
              className="btn btn-primary pl-5 pr-5 text-uppercase"
            >
              {submit}
            </button>
          </div>
        </div>
      </div>
    );
  }
}
/*
 * mapDispatchToProps
 */
const mapDispatchToProps = dispatch => ({
  getBillingAddress: () => dispatch(billingAddressAction()),
  selectAddress: id => dispatch(selectAddressAction(id))
});
/*
 * mapStateToProps
 */
const mapStateToProps = state => ({
  ...state
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SmartCheckout);
