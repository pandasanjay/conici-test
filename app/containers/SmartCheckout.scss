@import "../var.scss";

.sh-actions {
    border-top: 3px solid $secondry-color
}

.sh-add-card {
    cursor: pointer;
    border: 1px solid #000;
    background-color: #005578;
    color: #fff;
    border-radius: .5rem;
}

.sh-h3 {
    font-size: 24px;
    font-weight: bold;
}