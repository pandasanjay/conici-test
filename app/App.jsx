import React, { Component } from "react";
import { HashRouter as Router, Route } from "react-router-dom";
import SmartCheckout from "./containers/SmartCheckout";
import Layout from "./containers/Layout";

import "./global.scss";

export class App extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Router>
        <Layout>
          <Route path="/" component={SmartCheckout} />
        </Layout>
      </Router>
    );
  }
}

export default App;
