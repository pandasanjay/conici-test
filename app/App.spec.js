import React from "react";
import { shallow } from "enzyme";
import { HashRouter as Router, Route } from "react-router-dom";
import Layout from "./containers/Layout";
import SmartCheckout from "./containers/SmartCheckout";
import App from "./App";

describe("Test the App component", function() {
  let component;
  beforeEach(() => {
    component = shallow(<App />);
  });
  it("should have match with snapshot", function() {
    expect(component).toMatchSnapshot();
  });
  it("should have route", () => {
    expect(component.find(Router).length).toEqual(1);
    expect(component.find(Layout).length).toEqual(1);
    expect(component.find(Route).length).toEqual(1);
    expect(component.find(Route).props().path).toEqual("/");
    expect(component.find(Route).props().component).toEqual(SmartCheckout);
  });
});
