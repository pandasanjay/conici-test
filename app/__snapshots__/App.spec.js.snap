// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the App component should have match with snapshot 1`] = `
<HashRouter>
  <Connect(withRouter(Layout))>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(SmartCheckout)",
          "type": [Function],
        }
      }
      path="/"
    />
  </Connect(withRouter(Layout))>
</HashRouter>
`;
